src/main.ts

```diff
@@ -3,17 +3,46 @@ import { VIEW_TYPE_TODO } from './constants';
 import { TodoItemView, TodoItemViewProps } from './ui/TodoItemView';
 import { TodoItem, TodoItemStatus } from './model/TodoItem';
 import { TodoIndex } from './model/TodoIndex';
+import { DEFAULT_SETTINGS, TodoPluginSettings, TodoPluginSettingTab } from './settings';
 
 export default class TodoPlugin extends Plugin {
+  settings: TodoPluginSettings;
   private todoIndex: TodoIndex;
   private view: TodoItemView;
 
   constructor(app: App, manifest: PluginManifest) {
     super(app, manifest);
-    this.todoIndex = new TodoIndex(this.app.vault, this.tick.bind(this));
+    this.todoIndex = new TodoIndex(this.app.vault, this.tick.bind(this), this);
   }
 
   async onload(): Promise<void> {
+    await this.loadSettings();
+    this.addSettingTab(new TodoPluginSettingTab(this.app, this));
+
+    this.addCommand({
+      id: 'refresh-all',
+      name: 'Refresh All',
+      callback: () => {
+        this.prepareIndex(true);
+      },
+    });
```

Comment:
oh this is a great one! Maybe we can add one command to easily create a todo list. I like how easy to create a todo list at Craft, maybe we can bring it to obsidian
<img width="331" alt="Screenshot 2022-08-16 at 18 47 22" src="https://user-images.githubusercontent.com/1593488/184934394-346bc00a-b076-4aa1-a83d-f0272f5a41fe.png">